Sources/Core/JavaModelRenderer.swift

```diff
@@ -30,6 +30,10 @@ public struct JavaModelRenderer: JavaFileRenderer {
         }
     }
 
+    func renderStaticTypeString() -> JavaIR.Property {
+        return JavaIR.Property(annotations: [], modifiers: [.public, .static, .final], type: "String", name: "TYPE", initialValue: "\"" + className.lowercased() + "\"")
```

Comment:
The value of this variable should mirror what we use `polymorphicTypeIdentifier` for in Objective-C.

You should use the `typeIdentifier` getter on `SchemaObjectRoot` here since we allow the type strings to vary if `algebraicTypeIdentifier` is specified (which is intended to override the value of `type` in schemas if needed)

If you look at the ObjC implementation of `polymorphicTypeIdentifier` you will see it uses this getter method in `SchemaObjectRoot`
```
public struct SchemaObjectRoot: Equatable {
   ....
    var typeIdentifier: String {
        return algebraicTypeIdentifier ?? name
    }
}
```

Reply:
Thanks for the catch! Done.

Also ran `make integration_test` to update the samples.